spec/models/check_claim_spec.rb

```diff
@@ -0,0 +1,63 @@
+require "rails_helper"
+
+RSpec.describe CheckClaim, type: :model do
+  let(:user) { "123" }
+  let(:check_create) do
+    described_class.new(claim_id: claim.id, checked_by: user, status: status)
+  end
+
+  context "with a submittable claim" do
+    let(:claim) { create(:claim, :submitted) }
+
+    context "when a claim is approved" do
+      let(:status) { "approved" }
+
+      it "creates a check" do
+        expect { check_create.perform }.to change { Check.count }.by(1)
+        check = claim.reload.check
+        expect(check.checked_by).to eq(user)
+        expect(check.status).to eq(status)
+      end
+
+      it "queues an email job" do
+        expect { check_create.perform }
+          .to have_enqueued_job(ActionMailer::DeliveryJob)
+          .with("ClaimMailer", "approved", "deliver_now", claim)
+      end
+    end
+
+    context "when check is invalid" do
+      let(:status) { nil }
+      let(:user) { nil }
```

Comment:
Can we split these cases up? We want to make sure we can't create a check without either of them.